### ISSUE TITLE
Draft spec for gene-function search

### ISSUE THREAD

**Issue by StevenCannon-USDA** [legumeinfo/website-ui-specs#7]
Please see draft spec for gene function searches (for new gene_function collection type), here:
https://github.com/legumeinfo/website-ui-specs/tree/main/gene-function-search

Note: this spec puts the cart before the horse in the sense that the InterMine back-ends are not yet in place. The data spec is here: https://github.com/legumeinfo/datastore-specifications/tree/main/Genus/species/gene_functions

... and sample data [here for Glycine](https://data.legumeinfo.org/Glycine/max/gene_functions/).

Please provide feedback via this issue.
@sammyjava @That-Thing @maxglycine @jd-campbell @alancleary @adf-ncgr

**Comment by sammyjava** [comment]
I look forward to building it if it's not a summer intern. Haven't poked at it yet.

**Comment by StevenCannon-USDA** [comment]
@sammyjava Thanks. Implicit in this search is a gene-function page of some sort for the mines. Let me know what kind of input you will need for that. We're collecting the data into https://github.com/legumeinfo/gene-function-registry and then schlepping it over to the relevant gene_function area after review.

**Comment by sammyjava** [comment]
Correct, I'll need to modify the mine model and add a loader for these data. That will be in 5.1.0.4. I'm currently starting a 5.1.0.3 build run, with GlycineMine first. So you've got a cart leading the horse situation for a while.

**Comment by maxglycine** [comment]
Need to add a link to a list of traits in the database so the user has an idea of what traits are available in the gene function registry. This would be similar to what is found in the trait association search (https://www.soybase.org/tools/search/trait.html) but it should be specific to the gene function traits.  Maybe a search within the gene function data.
@StevenCannon-USDA @jd-campbell @That-Thing

**Comment by maxglycine** [comment]
@StevenCannon-USDA I did look at https://github.com/legumeinfo/website-ui-specs/tree/main/gene-function-search.  It was there that I realized that the input interface could be improved by a link to a list of known gene functions.  This was missing from the interface mockup you made.  

I was assigned sometime back to review the interface and make comments/improvements.  That is what I did.

It looks to me that my comment was placed in the correct place (gene_function_search) because that is where the interface mockup was.  My comment was not about what was in the gene_function_registry.
@jd-campbell @sammyjava @That-Thing @adf-ncgr @alancleary

**Comment by StevenCannon-USDA** [comment]
@maxglycine I like your suggestion.
I agree that it could be handled similarly to the page that's linked from the top of 
https://www.soybase.org/tools/search/trait.html
This wouldn't be part of the gene-function component, but could be added in the page that calls the component.
